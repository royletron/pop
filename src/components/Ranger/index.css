div.this {
}

div.title {
  font-weight: bold;
}

div.slider {
  width: 100%;
  height: 20px;
  background-color: #00B4FF;
  border: solid 1px #009AEF;
  border-radius: 10px;
}

div.sliderhandle {
  width: 30px;
  height: 20px;
  text-align: center;
  color: white;
  background-color: rgba(0, 0, 0, 0.3);
  border-radius: 10px;
}
