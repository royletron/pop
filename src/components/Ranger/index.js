import React from 'react';
import classNames from 'classnames';
import styles from './index.css';

import ReactSlider from 'react-slider';

export default class Ranger extends React.Component {
  constructor (props) {
    super(props);
    this.state = {
      min: props.min,
      max: props.max
    }
  }
  componentDidUpdate(prevProps, prevState) {
    if(prevProps !== this.props) {
      this.setState({
        min: this.state.min < this.props.min ? this.props.min : this.state.min,
        max: this.state.max > this.props.max ? this.props.max : this.state.max
      })
    }
  }
  onChange = () => {
    if(this.props.onChange) {
      var values = this.refs.slider.getValue();
      this.props.onChange(values[0], values[1])
      this.setState({
        min: values[0],
        max: values[1]
      })
    }
  }
  render () {
    return (
      <div style={this.props.style} className={classNames(styles.this)}>
        <div className={classNames(styles.title)}>{this.props.name}</div>
        <ReactSlider ref='slider' onChange={this.onChange} min={this.props.min} max={this.props.max} step={this.props.step || 1} className={classNames(styles.slider)} value={[this.state.min, this.state.max]}>
          <div className={classNames(styles.sliderhandle)}>{this.state.min}</div>
          <div className={classNames(styles.sliderhandle)}>{this.state.max}</div>
        </ReactSlider>
      </div>
    );
  }
}
