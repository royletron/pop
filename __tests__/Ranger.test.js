'use strict';

import React from 'react';
import Ranger from '../src/components/Ranger';
import renderer from 'react-test-renderer';

it('should display a ranger', () => {
  const component = renderer.create(
    <Ranger min={16} max={74} step={1} name='Age' />
  )
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})

it('should not update the ranger when min range is decreased', () => {
  const component = renderer.create(
    <Ranger min={16} max={74} step={1} name='Age' />
  )
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  component.update(<Ranger min={10} max={74} step={1} name='Age' />)
  expect(tree).toMatchSnapshot();
})

it('should update the ranger when min range is increased', () => {
  const component = renderer.create(
    <Ranger min={16} max={74} step={1} name='Age' />
  )
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  component.update(<Ranger min={22} max={74} step={1} name='Age' />)
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})

it('should not update the ranger when max range is increased', () => {
  const component = renderer.create(
    <Ranger min={16} max={74} step={1} name='Age' />
  )
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  component.update(<Ranger min={16} max={100} step={1} name='Age' />)
  expect(tree).toMatchSnapshot();
})

it('should update the ranger when max range is decreased', () => {
  const component = renderer.create(
    <Ranger min={16} max={74} step={1} name='Age' />
  )
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  component.update(<Ranger min={16} max={60} step={1} name='Age' />)
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})
