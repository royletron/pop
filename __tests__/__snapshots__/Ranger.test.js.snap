exports[`test should display a ranger 1`] = `
<div
  className=""
  style={undefined}>
  <div
    className="">
    Age
  </div>
  <div
    className=""
    onClick={[Function bound ]}
    onMouseDown={[Function bound ]}
    style={
      Object {
        "position": "relative"
      }
    }>
    <div
      className="handle handle-0 "
      onMouseDown={[Function bound ]}
      onTouchStart={[Function bound ]}
      style={
        Object {
          "left": "0px",
          "position": "absolute",
          "willChange": "",
          "zIndex": 1
        }
      }>
      <div
        className="">
        16
      </div>
    </div>
    <div
      className="handle handle-1 "
      onMouseDown={[Function bound ]}
      onTouchStart={[Function bound ]}
      style={
        Object {
          "left": "0px",
          "position": "absolute",
          "willChange": "",
          "zIndex": 2
        }
      }>
      <div
        className="">
        74
      </div>
    </div>
  </div>
</div>
`;

exports[`test should not update the ranger when max range is increased 1`] = `
<div
  className=""
  style={undefined}>
  <div
    className="">
    Age
  </div>
  <div
    className=""
    onClick={[Function bound ]}
    onMouseDown={[Function bound ]}
    style={
      Object {
        "position": "relative"
      }
    }>
    <div
      className="handle handle-0 "
      onMouseDown={[Function bound ]}
      onTouchStart={[Function bound ]}
      style={
        Object {
          "left": "0px",
          "position": "absolute",
          "willChange": "",
          "zIndex": 1
        }
      }>
      <div
        className="">
        16
      </div>
    </div>
    <div
      className="handle handle-1 "
      onMouseDown={[Function bound ]}
      onTouchStart={[Function bound ]}
      style={
        Object {
          "left": "0px",
          "position": "absolute",
          "willChange": "",
          "zIndex": 2
        }
      }>
      <div
        className="">
        74
      </div>
    </div>
  </div>
</div>
`;

exports[`test should not update the ranger when max range is increased 2`] = `
<div
  className=""
  style={undefined}>
  <div
    className="">
    Age
  </div>
  <div
    className=""
    onClick={[Function bound ]}
    onMouseDown={[Function bound ]}
    style={
      Object {
        "position": "relative"
      }
    }>
    <div
      className="handle handle-0 "
      onMouseDown={[Function bound ]}
      onTouchStart={[Function bound ]}
      style={
        Object {
          "left": "0px",
          "position": "absolute",
          "willChange": "",
          "zIndex": 1
        }
      }>
      <div
        className="">
        16
      </div>
    </div>
    <div
      className="handle handle-1 "
      onMouseDown={[Function bound ]}
      onTouchStart={[Function bound ]}
      style={
        Object {
          "left": "0px",
          "position": "absolute",
          "willChange": "",
          "zIndex": 2
        }
      }>
      <div
        className="">
        74
      </div>
    </div>
  </div>
</div>
`;

exports[`test should not update the ranger when min range is decreased 1`] = `
<div
  className=""
  style={undefined}>
  <div
    className="">
    Age
  </div>
  <div
    className=""
    onClick={[Function bound ]}
    onMouseDown={[Function bound ]}
    style={
      Object {
        "position": "relative"
      }
    }>
    <div
      className="handle handle-0 "
      onMouseDown={[Function bound ]}
      onTouchStart={[Function bound ]}
      style={
        Object {
          "left": "0px",
          "position": "absolute",
          "willChange": "",
          "zIndex": 1
        }
      }>
      <div
        className="">
        16
      </div>
    </div>
    <div
      className="handle handle-1 "
      onMouseDown={[Function bound ]}
      onTouchStart={[Function bound ]}
      style={
        Object {
          "left": "0px",
          "position": "absolute",
          "willChange": "",
          "zIndex": 2
        }
      }>
      <div
        className="">
        74
      </div>
    </div>
  </div>
</div>
`;

exports[`test should not update the ranger when min range is decreased 2`] = `
<div
  className=""
  style={undefined}>
  <div
    className="">
    Age
  </div>
  <div
    className=""
    onClick={[Function bound ]}
    onMouseDown={[Function bound ]}
    style={
      Object {
        "position": "relative"
      }
    }>
    <div
      className="handle handle-0 "
      onMouseDown={[Function bound ]}
      onTouchStart={[Function bound ]}
      style={
        Object {
          "left": "0px",
          "position": "absolute",
          "willChange": "",
          "zIndex": 1
        }
      }>
      <div
        className="">
        16
      </div>
    </div>
    <div
      className="handle handle-1 "
      onMouseDown={[Function bound ]}
      onTouchStart={[Function bound ]}
      style={
        Object {
          "left": "0px",
          "position": "absolute",
          "willChange": "",
          "zIndex": 2
        }
      }>
      <div
        className="">
        74
      </div>
    </div>
  </div>
</div>
`;

exports[`test should update the ranger when max range is decreased 1`] = `
<div
  className=""
  style={undefined}>
  <div
    className="">
    Age
  </div>
  <div
    className=""
    onClick={[Function bound ]}
    onMouseDown={[Function bound ]}
    style={
      Object {
        "position": "relative"
      }
    }>
    <div
      className="handle handle-0 "
      onMouseDown={[Function bound ]}
      onTouchStart={[Function bound ]}
      style={
        Object {
          "left": "0px",
          "position": "absolute",
          "willChange": "",
          "zIndex": 1
        }
      }>
      <div
        className="">
        16
      </div>
    </div>
    <div
      className="handle handle-1 "
      onMouseDown={[Function bound ]}
      onTouchStart={[Function bound ]}
      style={
        Object {
          "left": "0px",
          "position": "absolute",
          "willChange": "",
          "zIndex": 2
        }
      }>
      <div
        className="">
        74
      </div>
    </div>
  </div>
</div>
`;

exports[`test should update the ranger when max range is decreased 2`] = `
<div
  className=""
  style={undefined}>
  <div
    className="">
    Age
  </div>
  <div
    className=""
    onClick={[Function bound ]}
    onMouseDown={[Function bound ]}
    style={
      Object {
        "position": "relative"
      }
    }>
    <div
      className="handle handle-0 "
      onMouseDown={[Function bound ]}
      onTouchStart={[Function bound ]}
      style={
        Object {
          "left": "0px",
          "position": "absolute",
          "willChange": "",
          "zIndex": 1
        }
      }>
      <div
        className="">
        16
      </div>
    </div>
    <div
      className="handle handle-1 "
      onMouseDown={[Function bound ]}
      onTouchStart={[Function bound ]}
      style={
        Object {
          "left": "0px",
          "position": "absolute",
          "willChange": "",
          "zIndex": 2
        }
      }>
      <div
        className="">
        60
      </div>
    </div>
  </div>
</div>
`;

exports[`test should update the ranger when min range is increased 1`] = `
<div
  className=""
  style={undefined}>
  <div
    className="">
    Age
  </div>
  <div
    className=""
    onClick={[Function bound ]}
    onMouseDown={[Function bound ]}
    style={
      Object {
        "position": "relative"
      }
    }>
    <div
      className="handle handle-0 "
      onMouseDown={[Function bound ]}
      onTouchStart={[Function bound ]}
      style={
        Object {
          "left": "0px",
          "position": "absolute",
          "willChange": "",
          "zIndex": 1
        }
      }>
      <div
        className="">
        16
      </div>
    </div>
    <div
      className="handle handle-1 "
      onMouseDown={[Function bound ]}
      onTouchStart={[Function bound ]}
      style={
        Object {
          "left": "0px",
          "position": "absolute",
          "willChange": "",
          "zIndex": 2
        }
      }>
      <div
        className="">
        74
      </div>
    </div>
  </div>
</div>
`;

exports[`test should update the ranger when min range is increased 2`] = `
<div
  className=""
  style={undefined}>
  <div
    className="">
    Age
  </div>
  <div
    className=""
    onClick={[Function bound ]}
    onMouseDown={[Function bound ]}
    style={
      Object {
        "position": "relative"
      }
    }>
    <div
      className="handle handle-0 "
      onMouseDown={[Function bound ]}
      onTouchStart={[Function bound ]}
      style={
        Object {
          "left": "0px",
          "position": "absolute",
          "willChange": "",
          "zIndex": 1
        }
      }>
      <div
        className="">
        22
      </div>
    </div>
    <div
      className="handle handle-1 "
      onMouseDown={[Function bound ]}
      onTouchStart={[Function bound ]}
      style={
        Object {
          "left": "0px",
          "position": "absolute",
          "willChange": "",
          "zIndex": 2
        }
      }>
      <div
        className="">
        74
      </div>
    </div>
  </div>
</div>
`;
